packages/@orbit/jsonapi/src/lib/query-operators.ts

```diff
@@ -161,17 +190,19 @@ export const QueryOperators: Dict<QueryOperator> = {
     );
     requestProcessor.preprocessResponseDocument(document, query);
     const deserialized = requestProcessor.serializer.deserialize(document);
-    const relatedRecords = deserialized.data;
+    const relatedRecords = deserialized.data as Record[];
 
     const operations = requestProcessor.operationsFromDeserializedDocument(
       deserialized
     );
-    operations.push({
-      op: 'replaceRelatedRecords',
-      record,
-      relationship,
-      relatedRecords
-    } as ReplaceRelatedRecordsOperation);
+    for (let relatedRecord of relatedRecords) {
+      operations.push({
+        op: 'addToRelatedRecords',
+        record,
+        relationship,
+        relatedRecord
+      } as AddToRelatedRecordsOperation);
+    }
```

Comment:
If the results are not filtered or paginated, then I think that `replaceRelatedRecords` is still the best operation to emit.